src/shared_utilities/FileReader.jl

```diff
@@ -0,0 +1,371 @@
+"""
+    FileReader
+
+This module coordinates reading of input data from NetCDF files,
+and allows us to read in data containing a time component.
+It also includes regridding and temporal interpolations
+of this data.
+
+This is based on ClimaCoupler.jl's BCReader and TimeManager modules.
+"""
+module FileReader
+
+using ClimaComms
+using ClimaCore: Fields
+using Dates
+using JLD2
+import CFTime
+
+using ClimaLSM.Regridder
+
+export to_datetime,
+    datetime_to_strdate,
+    FileInfo,
+    file_info_init,
+    float_type_fi,
+    read_data_fields!,
+    next_date_in_file,
+    interpolate_data
+
+
+"""
+    to_datetime(date)
+
+Convert a DateTime-like object (e.g. DateTimeNoLeap) to a DateTime,
+using CFTime.jl. We need this since the CESM2 albedo file contains
+DateTimeNoLeap objects for dates, which can't be used for math with DateTimes.
+
+Note that this conversion may fail if the date to convert doesn't
+exist in the DateTime calendar.
+
+# Arguments
+- `date`: object to be converted to DateTime
+"""
+to_datetime(date) = CFTime.reinterpret(DateTime, date)
+
+"""
+    datetime_to_strdate(datetime::DateTime)
+
+Convert from Date to String ("YYYYMMDD") format.
+
+# Arguments
+- `datetime`: [DateTime] object to be converted to string
+"""
+datetime_to_strdate(datetime::DateTime) =
+    string(lpad(Dates.year(datetime), 4, "0")) *
+    string(string(lpad(Dates.month(datetime), 2, "0"))) *
+    string(lpad(Dates.day(datetime), 2, "0"))
+
+"""
+    FileInfo
+
+Stores information specific to each boundary condition from a file and each variable.
+
+# Inputs:
+- file_path::S                    # directory of the data file
+- comms::X                        # communication context used for MPI
+- varname::V                      # name of the variable
```

Comment:
should `varname` be plural here? (I imagine there's more than one variable per file)

Reply:
So far we've only been using files that have one variable we want to read in. We could extend this in the future if we want to use a file that has multiple variables.